web-app/js/portal/filter/Filter.js

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2015 IMOS
+ *
+ * The AODN/IMOS Portal is distributed under the terms of the GNU General Public License
+ *
+ */
+
+Ext.namespace('Portal.filter');
+
+Portal.filter.Filter = Ext.extend(Object, {
+
+    constructor: function(filterDetail) {
+
+        this.name = filterDetail.name;
+        this.displayLabel = filterDetail.label;
+        this.type = filterDetail.type;
+        this.visualised = filterDetail.visualised;
+        this.wmsStartDateName = filterDetail.wmsStartDateName;
+        this.wmsEndDateName = filterDetail.wmsEndDateName;
+    },
+
+    setValue: function(value) {
```

Comment:
I hated the repetition here so in a refactoring (which isn't on this branch) I pulled this all out. See https://github.com/aodn/aodn-portal/blob/ee69e19157de6b36b4d0bcedf57a265230343c5d/web-app/js/portal/filter/Filter.js#L14

So if we were to remove the accessor methods now there wouldn't be any indication of which fields are going to be present other than by searching other code for their usages. Do you think that would be a problem?